plugins/lookup/rhub_api.py

```diff
@@ -0,0 +1,79 @@
+__metaclass__ = type
+
+
+DOCUMENTATION = """
+module: rhub_api
+short_description: Interacts with Resource Hub API.
+description:
+    - Retrieve data from Resource Hub API.
+options:
+    _terms:
+      description:
+        - API endpoint path.
+      required: true
+      type: str
+    addr:
+      description:
+        - Resource Hub base adress.
+      required: true
+      type: str
+      env:
+        - name: RHUB_API_ADDR
+    username:
+      description:
+        - API user.
+      required: true
+      type: str
+      env:
+        - name: RHUB_API_USER
+    password:
+      description:
+        - Password for API user.
+      required: true
+      type: str
+      env:
+        - name: RHUB_API_PASS
+"""
+
+EXAMPLES = """
+- name: get cluster info
+  ansible.builtin.debug:
+    msg: "{{ lookup('rhub.rhub.rhub_api', '/v0/lab/cluster/1 addr=https://rhub.stage.example.com/') }}"
+"""
+
+RETURN = """
```

Comment:
Can we have an option to return results in different formats such as JSON?

Reply:
API always returns JSON and that is automatically parsed by ansible.

```yaml
    - name: get cluster info
      set_fact:
        cluster_data: '{{ lookup("rhub.rhub.rhub_api", "/v0/lab/cluster/{{ cluster_id }}") }}'
```

```
TASK [get cluster info] ****************************************************************************************************************
ok: [localhost] => {
    "ansible_facts": {
        "cluster_data": {
            "created": "2021-08-25T11:48:54.483830+00:00",
            "description": "test",
            "group_id": null,
            "id": 1,
            "lifespan_expiration": null,
            "name": "testest0",
            "quota": {
                "num_vcpus": 40,
                "num_volumes": 40,
                "ram_mb": 200000,
                "volumes_gb": 540
            },
            "region_id": 2,
            "reservation_expiration": null,
            "status": null,
            "user_id": "55b52d88-1c48-419c-96b7-b6ddac6a0d78"
        }
    },
    "changed": false
}
```